HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,268 @@
+// -*- C++ -*-
+//
+// Package:    HLTrigger/HLTHFJetCleaner
+// Class:      HLTHFJetCleaner
+//
+/**\class HLTHFJetCleaner HLTHFJetCleaner.cc HLTrigger/HLTHFJetCleaner/plugins/HLTHFJetCleaner.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Alp Akpinar
+//         Created:  Wed, 01 Dec 2021 15:47:19 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/METReco/interface/CaloMET.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+//
+// class declaration
+//
+
+class HLTHFJetCleaner : public edm::stream::EDProducer<> {
+public:
+  typedef std::vector<reco::PFJet> JetCollection;
+  explicit HLTHFJetCleaner(const edm::ParameterSet&);
+  ~HLTHFJetCleaner() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  template <typename T>
+  std::vector<T> extractValueMapParameters(edm::Handle<edm::ValueMap<T>> valueMapHandle, size_t numJets);
+
+  edm::EDGetTokenT<edm::View<reco::PFJet>> jetToken_;
+  edm::EDGetTokenT<edm::View<reco::CaloMET>> metToken_;
```

Comment:
Same comment on (Calo)MET regarding templating. Does this needs to be CaloMET? Or can it be the base class reco::MET ? Or should this be templated to support also other objects like reco::PFMET?

It looks like reco::MET would be sufficient here.

Reply:
Similar to the jets case, this should be updated to be the base class `reco::MET` if that works.

Reply:
Implemented in [2a4e694](https://github.com/cms-sw/cmssw/pull/37539/commits/2a4e6947671e4c8730dd5dd4c1813d74ee793c1c)